openreview/tools.py

```diff
@@ -324,38 +324,43 @@ def replace_members_with_ids(client, group):
     group.members = ids + emails
     client.post_group(group)
 
-def _get_all(get_function, invitation, limit):
+def iterget(get_function, **kwargs):
     '''
-    Internal helper function for retrieving all objects of a certain type without
-    needing to set the limit in the API call.
-
+    Iterator over a given get function from the client.
     '''
     done = False
-    objects = []
     offset = 0
+    params = {
+        'limit': 1000 if 'limit' not in kwargs else kwargs['limit']
+    }
+
     while not done:
-        batch = get_function(invitation=invitation, limit=limit, offset=offset)
-        objects += batch
-        offset += limit
-        if len(batch) < limit:
+        params['offset'] = offset
+
+        params.update(kwargs)
+        batch = get_function(**params)
+        offset += params['limit']
+        for obj in batch:
+            yield obj
+
+        if len(batch) < params['limit']:
             done = True
-    return objects
 
 def get_all_tags(client, invitation, limit=1000):
     '''
     Given an invitation, returns all Tags that respond to it, ignoring API limit.
     '''
-    return _get_all(client.get_tags, invitation, limit)
+    return list(iterget(client.get_tags, invitation=invitation, limit=limit))
```

Comment:
Does it work in Python? I haven't tried it.

Reply:
May be you can add a test :)

Reply:
yeah it does. speaking of which, should I be adding a test for this? I'm not sure how we've been doing it